network/ipmasq.go

```diff
@@ -57,3 +57,17 @@ func setupIPMasq(ipn ip.IP4Net) error {
 
 	return nil
 }
+
+func teardownIPMasq(ipn ip.IP4Net) error {
+	ipt, err := iptables.New()
+	if err != nil {
+		return fmt.Errorf("failed to setup IP Masquerade. iptables was not found")
+	}
+
+	err = ipt.DeleteChain("nat", "FLANNEL")
```

Comment:
I see what you mean -- the chain must be empty before deletion. I think you're right, not sure how it seemed to work, I'll investigate.